changelogs/unreleased/fix-fkey-constraint-check-memleak.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/box
+
+- Fixed a memory leak on foreign key constraint check failure.
```

Comment:
Again, this looks critical. Please file an issue.

Reply:
Created an issue #10476